### PR TITLE
Remove `HAVE_LARGE_INSN_LIMIT`

### PR DESCRIPTION
**Pull request by dylandreimerink**
Now that #30456 is completed we have raised the minimum supported kernel version to v5.4 or RHEL8.6. This means that we can now assume support for certain features, one of these is the large instruction limit. Since #30896 added an assert of the large instruction limit on startup, we can remove all other mentions of it in the codebase, which is a pleasant cleanup.

### CODE REVIEW

File: bpf/include/bpf/verifier.h [hunk 1]
```diff
@@ -12,9 +12,7 @@
  */
 static __always_inline void relax_verifier(void)
 {
-#ifndef HAVE_LARGE_INSN_LIMIT
        volatile int __maybe_unused id = get_smp_processor_id();
```
Comment (julianwiedmann) [inline bpf/include/bpf/verifier.h]: Note that this is a `#ifndef` !

Comment (dylandreimerink) [reply bpf/include/bpf/verifier.h]: Oh, shit :facepalm:

**Review by julianwiedmann — APPROVED** [review]
ty!

**Review by rgo3 — APPROVED** [review]

**Review by markpash — APPROVED** [review]

**Comment by dylandreimerink** [comment]
@danehans PTAL

File: daemon/cmd/kube_proxy_replacement.go [hunk 2]
```diff
@@ -448,8 +448,7 @@ func finishKubeProxyReplacementInit(sysctl sysctl.Sysctl) error {
 
 	option.Config.NodePortNat46X64 = option.Config.IsDualStack() &&
 		option.Config.DatapathMode == datapathOption.DatapathModeLBOnly &&
-		option.Config.NodePortMode == option.NodePortModeSNAT &&
-		probes.HaveLargeInstructionLimit() == nil
```
Comment (danehans) [inline daemon/cmd/kube_proxy_replacement.go:452]: Should `probes.HaveLargeInstructionLimit()` be removed if it's no longer being called throughout the codebase?

**Review by danehans — APPROVED** [review]
I have one non-blocking question.

File: daemon/cmd/kube_proxy_replacement.go (hunk 2 above)
Comment (dylandreimerink) [reply daemon/cmd/kube_proxy_replacement.go:452]: No. We still use the probe, but only in `CheckRequirements` in `pkg/datapath/linux/requirements.go`. Which is run during startup to assert that we have a proper kernel.

**Pull request merged**